vcr/credential/validator.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/nuts-foundation/go-did"
+)
+
+// Validator is the interface specific VC verification.
+// Every VC will have it's own rules of verification.
+type Validator interface {
+	// Validate the given credential according to the rules of the VC type.
+	Validate(credential did.VerifiableCredential) error
+}
+
+// validate the default fields
+func validate(credential did.VerifiableCredential) error {
+
+	if !containsType(credential, DefaultCredentialType) {
+		return errors.New("validation failed: 'VerifiableCredential' is required")
```

Comment:
Introduce a new `ErrInvalid = errors.New("validation failed")` error?

Reply:
Added a custom validationError that also implements `Is(error) bool` so a check can be done like:
```
if errors.Is(err, ValidationError) {

}
```